File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        left= 0
        longest = set()
        max_len = 0
        for i in range(len(s)):
            while s[i] in longest:
                longest.remove(s[left])
                left+=1
            longest.add(s[i])
            max_len = max(max_len, len(longest))
            
        return max_len
 

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let s_set = new Set();
    let left = 0;
    let longSub = 0;
    for (let right = 0; right < s.length; right++){
        while (s_set.has(s[right])){
            s_set.delete(s[left])
            left ++;
        }
        s_set.add(s[right])
        longSub = Math.max(right - left+1, longSub);
        
    }
    return longSub;
};